draft-ietf-tls-extended-key-update.md

```diff
@@ -797,9 +803,13 @@ This section discusses additional security and operational aspects introduced by
 
 Extended Key Update assumes a transient compromise of the current application
 traffic keys, not a persistent attacker with ongoing access to key material.
-Long-term private keys are assumed secure, and post-compromise security
-therefore remains achievable. This represents a short-term exposure of system
-memory or TLS session state after the handshake has completed.
+Long-term private keys are assumed to remain secure, as they are stored in a
+secure element, such as a Trusted Execution Environment (TEE), Hardware Security
+Module (HSM), or Trusted Platform Module (TPM). In contrast, application traffic
+keys are stored in within the rich operating system, where short-term exposure due
+to memory disclosure or transient compromise may occur. Post-compromise security
+therefore remains achievable, as a fresh key exchange during an Extended Key Update
+restores confidentiality once the compromise is resolved.
```

Comment:
I think post-compromise security breaks if adversary gets both current application traffic keys (which are keys most at risk of compromise, because older keys tend to get overwritten). MITM adversary can use the keys to initiate EKU with both peers, and then compute the resulting application traffic keys. The keys will not match, but adversary is able to re-encrypt the traffic. Furthermore, EKU can not restore security because adversary can re-write the shares, like how one does MITM on unauthenticated Diffie-Hellman.